internal/rbd/controllerserver.go

```diff
@@ -833,6 +833,16 @@ func (cs *ControllerServer) DeleteVolume(
 	}
 	defer cs.OperationLocks.ReleaseDeleteLock(volumeID)
 
+	if isMigrationVolHandle(volumeID) {
+		log.DebugLog(ctx, "migration volume handle : %s and secrets: %v", volumeID, req.GetSecrets())
+		err = parseAndDeleteMigratedVolume(ctx, volumeID, cr)
+		if err != nil {
+			return nil, err
```

Comment:
parseAndDeleteMigratedVolume is not returning GRPC error code/message in all error cases. please make sure it returns proper GRPC errors or return a GRPC error from here